workbench-core/authorization/src/dynamicAuthorization/dynamicAuthorizationService.ts

```diff
@@ -123,12 +123,32 @@ export class DynamicAuthorizationService {
    * @throws {@link TooManyRequestsError} - too many requests error
    */
   public async createGroup(createGroupRequest: CreateGroupRequest): Promise<CreateGroupResponse> {
-    const response = await this._groupManagementPlugin.createGroup(createGroupRequest);
-    await this._groupManagementPlugin.setGroupStatus({
-      groupId: createGroupRequest.groupId,
-      status: 'active'
-    });
-    return response;
+    const metadata: Metadata = {
+      actor: createGroupRequest.authenticatedUser,
+      action: this.createGroup.name,
+      source: {
+        serviceName: DynamicAuthorizationService.name
+      },
+      requestBody: createGroupRequest
+    };
+
+    try {
+      const response = await this._groupManagementPlugin.createGroup(createGroupRequest);
+      await this._groupManagementPlugin.setGroupStatus({
+        groupId: createGroupRequest.groupId,
+        status: 'active'
+      });
+
+      metadata.statusCode = 200;
+      await this._auditService.write(metadata, response);
+
+      return response;
+    } catch (error) {
+      metadata.statusCode = 400;
```

Comment:
Is this status code accurate? TooManyRequestsError should probably be a 429. If the error is on our end such as the user pool or db being off line, this should be a 5XX as appropriate.

Reply:
I think the original design had `success` and `failure` and the possible statuses, they were changed to `200` and `400` at some point. @htztomic might have more context

Reply:
understood. Thanks!